src/reanimated2/layoutReanimation/animationBuilder/Keyframe.ts

```diff
@@ -72,8 +73,8 @@ class InnerKeyframe implements IEntryExitAnimationBuilder {
       Initialize parsedKeyframes for properties provided in initial keyframe
     */
     Object.keys(initialValues).forEach((styleProp: string) => {
-      if (styleProp === 'transform') {
-        initialValues[styleProp]?.forEach((transformStyle, index) => {
+      if (styleProp === 'transform' && Array.isArray(initialValues.transform)) {
```

Comment:
Shouldn't we split it into separate conditionals?
```suggestion
      if (styleProp === 'transform') {
        if (Array.isArray(initialValues.transform)) {
```